File: app/src/main/java/ceui/lisa/whitefox/Container.java
package ceui.lisa.whitefox;


import com.franmontiel.persistentcookiejar.ClearableCookieJar;
import com.franmontiel.persistentcookiejar.PersistentCookieJar;
import com.franmontiel.persistentcookiejar.cache.SetCookieCache;
import com.franmontiel.persistentcookiejar.persistence.SharedPrefsCookiePersistor;

import okhttp3.OkHttpClient;

public class Container {

    private OkHttpClient mOkHttpClient;

    public OkHttpClient getOkHttpClient() {
        if (mOkHttpClient == null) {
            ClearableCookieJar cookieJar = new PersistentCookieJar(new SetCookieCache(),
                    new SharedPrefsCookiePersistor(App.context));
            mOkHttpClient = new OkHttpClient.Builder()
                    .cookieJar(cookieJar)
                    .build();
        }
        return mOkHttpClient;
    }

    private Container() {
    }

    private static final class Holder {
        private static final Container INSTANCE = new Container();
    }

    public static Container get() {
        return Holder.INSTANCE;
    }
}

File: app/src/main/java/ceui/lisa/whitefox/Container.kt
package ceui.lisa.whitefox

import com.franmontiel.persistentcookiejar.ClearableCookieJar
import com.franmontiel.persistentcookiejar.PersistentCookieJar
import com.franmontiel.persistentcookiejar.cache.SetCookieCache
import com.franmontiel.persistentcookiejar.persistence.SharedPrefsCookiePersistor
import okhttp3.OkHttpClient

object Container {

    var client: OkHttpClient

    init {
        val cookieJar: ClearableCookieJar = PersistentCookieJar(
            SetCookieCache(),
            SharedPrefsCookiePersistor(App.context)
        )
        client = OkHttpClient.Builder()
            .cookieJar(cookieJar)
            .build()
    }
}